contracts/Refunder.sol

```diff
@@ -69,7 +76,17 @@ contract Refunder is
      */
     modifier onlySupportedParams(address targetContract, bytes4 identifier) {
         require(tx.gasprice <= maxGasPrice, "Gas price is too expensive");
-        require(refundables[targetContract][identifier], "It's not refundable");
+        require(refundables[targetContract][identifier].isSupported, "It's not refundable");
+
+        if(refundables[targetContract][identifier].validationContract != address(0)) {
+            bytes memory data = abi.encodeWithSelector(refundables[targetContract][identifier].validationFunc, msg.sender);
+            (bool success, bytes memory returnData) = refundables[targetContract][identifier].validationContract.call(data);
+
+            (bool decodedResult) = abi.decode(returnData, (bool));
+            
+            require(success, "Validation contract was thrown error");
+            require(decodedResult, "Msg.sender is not permitted to be refunded");
+        }
```

Comment:
I think that it is better if we place this logic inside the body of the function and not the `modifier`. Not only it will be more readable, but this way we will be sure that we have reentrancy protection as-well